public/app/features/dashboard-scene/panel-edit/getPanelFrameOptions.tsx

```diff
@@ -121,41 +125,69 @@ export function PanelFrameTitleInput({ panel, isNewElement }: { panel: VizPanel;
       ref={ref}
       data-testid={selectors.components.PanelEditor.OptionsPane.fieldInput('Title')}
       value={title}
-      onChange={(e) => setPanelTitle(panel, e.currentTarget.value)}
+      onFocus={() => setPrevTitle(title)}
+      onBlur={() => editPanelTitleAction(panel, title, prevTitle)}
+      // The full action (that can be undone) is done by setPanelTitle,
+      // But to see changes in the input field, canvas and outline we change the real value here
+      onChange={(e) => updatePanelTitleState(panel, e.currentTarget.value)}
     />
   );
 }
 
 export function PanelDescriptionTextArea({ panel }: { panel: VizPanel }) {
   const { description } = panel.useState();
+  const [prevDescription, setPrevDescription] = React.useState(panel.state.description);
 
   return (
     <TextArea
       id="description-text-area"
       value={description}
-      onChange={(e) => panel.setState({ description: e.currentTarget.value })}
+      onChange={(evt) => panel.setState({ description: evt.currentTarget.value })}
+      onFocus={() => setPrevDescription(panel.state.description)}
+      onBlur={() => {
```

Comment:
As a user. I find this to be a little bit confusing. WDYT about debouncing the action in `onChange`?

Reply:
I would not do that, recoding the action on debounce would be way more confusing and inconsistent vs recoding the change as a single action and not depend on how fast or slow you type